app/gui/src/project-view/components/visualizations/TableVisualization/TableVizDataSourceUtils.ts

```diff
@@ -0,0 +1,183 @@
+    case 'Mixed': {
+      const items = value.value.map((val: { valueType: ValueTypes; value: string }) =>
+        parseSingleArgument(val, tempModule),
+      )
+      return Ast.Vector.new(tempModule, items)
+    }
+    default:
+      return Ast.parseExpression(value, tempModule)!
+  }
+}
+
+export const parseArgument = (arg: Argument, tempModule: Ast.MutableModule) => {
+  if (Array.isArray(arg)) {
+    const itemList = arg.map((i) => {
+      return typeof i === 'string' ?
+          Ast.parseExpression(i, tempModule)!
+        : parseSingleArgument(i, tempModule)
+    })
+    return Ast.Vector.new(tempModule, itemList!)
+  }
+  return Ast.parseExpression(arg, tempModule)!
+}
+
+export const convertSortModel = (request: IServerSideGetRowsRequest, columnHeaders: string[]) => {
+  const sortColIndexesMap = request.sortModel.map((sortCol) => {
+    return `${columnHeaders.findIndex((h: string) => sortCol.colId === h)}`
+  })
+  const sortColIndexes = sortColIndexesMap.length ? sortColIndexesMap : 'Nothing'
+  const sortDirections =
+    sortColIndexesMap.length ?
+      request.sortModel.map((sortCol) => {
+        return sortDirectionMap[sortCol.sort as SortDirection]
+      })
+    : 'Nothing'
+  return { sortColIndexes, sortDirections }
+}
+
+export const convertFilterModel = (
+  request: IServerSideGetRowsRequest,
+  columnHeaders: string[],
+  colTypeMap: Map<string, string>,
+) => {
+  const gridFilterModelList: Array<GridFilterModel> =
+    request.filterModel ? makeFilterModelList(request.filterModel) : []
+
+  const filterColumnNames = gridFilterModelList.map((filter) => filter.columnName)
+
+  const filterColumnIndexList =
+    filterColumnNames.length ?
+      filterColumnNames.map((colName) => `${columnHeaders.findIndex((h: string) => colName === h)}`)
+    : 'Nothing'
+
+  const filterActions =
+    filterColumnNames.length ?
+      gridFilterModelList.map((filter) => {
+        return filter.filterType === 'set' ?
+            '..Is_In'
+          : actionMap[filter.filterAction as FilterAction]
+      })
+    : 'Nothing'
+
+  const valueMap = gridFilterModelList.map((filter) => {
+    return {
+      valType: colTypeMap.get(filter.columnName),
+      action: actionMap[filter.filterAction as FilterAction],
+      value: getFilterValue(filter),
+    }
+  })
+
+  const valueList: Argument =
+    valueMap.length ?
+      valueMap.map((value) => {
+        if (value.valType === 'Mixed' && Array.isArray(value.value)) {
+          const parseValues = value.value.map((val) => {
+            return { valueType: getCellValueType(val), value: val }
+          })
+          return { valueType: 'Mixed', value: parseValues } as ValueTypeArgumentParent
+        }
+
+        if (
+          value.action === '..Between' &&
+          typeof value.value === 'object' &&
+          'fromValue' in value.value
+        ) {
+          return { valueType: value.valType as ValueTypes, value: `${value.value.fromValue}` }
```

Comment:
Could we pass `fromValue` and `toValue` as a pair here and avoid constructing the `toValueList`? Or is there some reason why this was problematic?